handler.c

```diff
@@ -1176,6 +1176,102 @@ static int alternative_handler(struct Body *a, struct State *s)
   return rc;
 }
 
+/**
+ * multilingual_handler - Parse a multi-lingual email
+ * @param a Body of the email
+ * @param s State for the file containing the email
+ * @retval 0 Always
+ */
+static int multilingual_handler(struct Body *a, struct State *s)
+{
+  struct Body *choice = NULL;
+  struct Body *b = NULL;
+  bool mustfree = false;
+  int rc = 0;
+  struct Body *first_part = NULL;
+  struct Body *zxx_part = NULL;
+  char *lang;
+
+  mutt_debug(2, "RFC8255 >> entering in handler multilingual handler\n");
+  if ((a->encoding == ENCBASE64) || (a->encoding == ENCQUOTEDPRINTABLE) ||
+      (a->encoding == ENCUUENCODED))
+  {
+    struct stat st;
+    mustfree = true;
+    fstat(fileno(s->fpin), &st);
+    b = mutt_body_new();
+    b->length = (long) st.st_size;
+    b->parts = mutt_parse_multipart(
+        s->fpin, mutt_param_get(&a->parameter, "boundary"), (long) st.st_size,
+        (mutt_str_strcasecmp("digest", a->subtype) == 0));
+  }
+  else
+    b = a;
+
+  a = b;
+
+  if (a->parts)
+    b = a->parts;
+  else
+    b = a;
+
+  mutt_debug(2, "RFC8255 >> preferred_languages set in config to '%s'\n", PreferredLanguages);
+  char *preferred_languages = mutt_str_strdup(PreferredLanguages);
+  lang = strtok(preferred_languages, ",");
+
+  while (lang)
+  {
+    while (b)
+    {
+      if (mutt_can_decode(b))
+      {
+        if (!first_part)
+          first_part = b;
+
+        if (b->language && (mutt_str_strcmp("zxx", b->language) == 0))
```

Comment:
I think this was introduced for testing and should be removed.

Reply:
No, there is indeed a "zxx" part, which is a "fallback" if the MUA cannot interpret that email at all. Please see RFC8255.